pydantic/main.py

```diff
@@ -134,6 +142,14 @@ def hash_func(self_: Any) -> int:
             )
 
             cls.__pydantic_model_complete__ = False  # Ensure this specific class gets completed
+
+            # preserve `__set_name__` protocol defined in https://peps.python.org/pep-0487
```

Comment:
is it okay to remove this?

Reply:
(I'm assuming you mean the actual code, not just this comment.)

There was an issue where it was added where someone specifically wanted to be able to use custom descriptors. (See https://github.com/pydantic/pydantic/pull/4407 and https://github.com/pydantic/pydantic/pull/4410). And there is a test below that requires it (`test_set_name` in test_create_model.py). And I tried adding `model_config = dict(non_field_types=(class_deco,))` to that test and it still failed without this code.

Is that a good enough reason to keep it, or would you still like to remove it?